### PR TITLE
Make CreateThreadPool Settable

### PR DESCRIPTION
**Pull request by ncteisen**

### CODE REVIEW

File: src/cpp/server/create_default_thread_pool.cc [hunk 1]
```diff
@@ -24,12 +24,18 @@
 
 namespace grpc {
 
-ThreadPoolInterface* CreateDefaultThreadPool() {
+static ThreadPoolInterface* CreateDefaultThreadPoolImpl() {
```
Comment (vjpai) [inline src/cpp/server/create_default_thread_pool.cc]: Prefer anonymous namespace to static, I think.

File: src/cpp/server/create_default_thread_pool.cc [hunk 2]
```diff
@@ -23,13 +23,22 @@
...
-ThreadPoolInterface* CreateDefaultThreadPool() {
+ThreadPoolInterface* CreateDefaultThreadPoolImpl() {
   int cores = gpr_cpu_num_cores();
   if (!cores) cores = 4;
   return new DynamicThreadPool(cores);
 }
 
+CreateThreadPoolFunc g_ctp_impl = CreateDefaultThreadPoolImpl;
+
+} // anon namespace
```
Comment (yang-g) [inline src/cpp/server/create_default_thread_pool.cc]: remove "anon "

Comment (ncteisen) [inline src/cpp/server/create_default_thread_pool.cc]: done

Comment (yang-g) [inline src/cpp/server/create_default_thread_pool.cc]: LGTM

**Comment by ncteisen** [comment]
Needs another LGTM from @vjpai as per Github rules

**Comment by ncteisen** [comment]
@vjpai ping

**Pull request merged**